lib/types/api.d.ts

```diff
@@ -0,0 +1,41 @@
+interface IOption<T> {
+    [key: string]: T
+}
+
+interface IPayload {
+    url: string,
+    params: IOption<any>
+}
+
+export interface IRazorpayHeaders {
+    'X-Razorpay-Account'?: string,
+    'Content-Type'?: string
+}
+
+/**
+ * Key-value pairs 
+ */
+export interface IMap<T> {
+    [key: string]: T | null;
+}
+
+export interface IRazorpayQuery {
+    from?: number;
+    to?: number;
+    count?: number;
+    skip?: number;
+}
+
+declare class API {
+    constructor(options: IOption<any>)
+    get(params: IPayload): Promise<any>
+    post(params: IPayload): Promise<any>
+    put(params: IPayload): Promise<any>
+    patch(params: IPayload): Promise<any>
+    delete(params: IPayload): Promise<any>
+    private allowedHeaders;
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Lets remove these since these are internal and not exposed to the user

Reply:
Removed all internal functions